<launch>

    <!-- bag的地址与名称 -->
    <arg name="bag_filename" default="/home/cw/Slam/bag_flies/kitti_2011_10_03_drive_0027_synced.bag"/>
    <!-- <arg name="bag_filename" default="/home/cw/Slam/bag_flies/aces.bag"/> -->
    <!-- 使用bag的时间戳 -->
    <param name="use_sim_time" value="true" />
    <param name="scan_topic" value="kitti/velo/pointcloud"/>
    <param name="lidarMsg_type" value="PointCloud2"/>
    <param name="cloud_topic" value="pretreat_cloud"/>
    <param name="laser_frame" value="velo_link"/>
    <param name="odom_frame" value="world"/>
    <!-- 启动节点 -->
    <node name="mrobot_frame_data_pretreat_node" pkg="mrobot_frame" type="mrobot_frame_data_pretreat_node" output="screen" />
    <node name="mrobot_frame_front_end_node" pkg="mrobot_frame" type="mrobot_frame_front_end_node" output="screen" />
    <!-- <node name="mrobot_frame_back_end_node" pkg="mrobot_frame" type="mrobot_frame_back_end_node" output="screen" /> -->
    <!-- <node name="mrobot_frame_loop_closing_node" pkg="mrobot_frame" type="mrobot_frame_loop_closing_node" output="screen" /> -->
    <!-- <node name="mrobot_frame_viewer_node" pkg="mrobot_frame" type="mrobot_frame_viewer_node" output="screen" /> -->
    <!-- <node name="mrobot_frame_mapping_node" pkg="mrobot_frame" type="mrobot_frame_mapping_node" output="screen" /> -->
    
    <!-- launch rviz -->
    <node name="rviz" pkg="rviz" type="rviz" required="false" args="-d $(find mrobot_frame)/rviz/front_end_3d.rviz" />
    <!-- play bagfile -->
    <node name="playbag" pkg="rosbag" type="play" args="--clock $(arg bag_filename) -r 0.5" />
    

</launch>